runtime/cudaq/qis/state.cpp

```diff
@@ -48,12 +48,12 @@ void state::dump(std::ostream &os) const { internal->dump(os); }
 
 std::complex<double>
 state::operator()(const std::initializer_list<std::size_t> &indices,
-                  std::size_t tensorIdx) {
+                  std::size_t tensorIdx) const {
   std::vector<std::size_t> idxVec(indices.begin(), indices.end());
   return (*internal)(tensorIdx, idxVec);
 }
 
-std::complex<double> state::operator[](std::size_t idx) {
+std::complex<double> state::operator[](std::size_t idx) const {
```

Comment:
@1tnguyen I had to add a const to a couple of methods there to allow accessing state elements in ArgumentConversion.cpp. Does not seem to break anything...